hepdata/modules/records/assets/js/hepdata_tables.js

```diff
@@ -95,17 +126,55 @@ HEPDATA.table_renderer = {
         // Initiates rendering of both related DOI table areas
         HEPDATA.table_renderer.render_related_dois(table_data.related_tables, "#related-tables");
         HEPDATA.table_renderer.render_related_dois(table_data.related_to_this, "#related-to-this-tables");
-
         HEPDATA.table_renderer.render_keywords(table_data.keywords, "#table_keywords");
-
+        $("#hepdata_table_loader").addClass("hidden");
+        $("#hepdata_table_content").removeClass("hidden");
+        // We also need to clear the figure
+        $("#figures").html('');
+
+        // Check that the table is both empty, and is larger than an empty table (bytes)
+        if(table_data.size > HEPDATA.size_load_check_threshold) {
+          // Set up filesize attempt section
+          $("#hepdata_table_loader").addClass("hidden");
+          var megabyte_size = (table_data.size / (1024 * 1024)).toFixed(2);
+          var additional_size = (HEPDATA.size_load_check_threshold / (1024 * 1024)).toFixed(2);
+          var message = "Table size is: " + megabyte_size + " MB, which is larger than our threshold of " + additional_size + " MB.";
+          d3.select("#filesize_table_size").html(message);
+          $("#hepdata_filesize_loader").removeClass("hidden");
+          $("filesize_table_confirm").removeClass("hidden");
+        }
+        else {
+          HEPDATA.table_renderer.display_table(url,
+            '#data_table_region',
+            '#data_visualization_region', load_all);
+        }
+      },
+      error: function (data, error) {
+        console.error(error);
+        d3.select("#hepdata_table_loading_failed_text").html('Failed to load table defined by ' + url);
```

Comment:
The other approach could be to have a library of possible messages to reduce the tendency towards magic strings. The current approach would make it very difficult if HEPData ever needs to be localized to other languages! But, if this is the way things are currently done, then stick with the existing standard.